abn/grpc/abn.proto

```diff
@@ -0,0 +1,51 @@
+// protoc --go_out=. --go_opt=paths=source_relative     --go-grpc_out=. --go-grpc_opt=paths=source_relative     abn/grpc/abn.proto
+// python -m grpc_tools.protoc -I../../../iter8-tools/iter8/abn/grpc --python_out=. --grpc_python_out=. ../../../iter8-tools/iter8/abn/grpc/abn.proto 
+
+syntax = "proto3";
+
+option go_package = "github.com/iter8-tools/iter8/abn/grpc";
+
+import "google/protobuf/empty.proto";
+package main;
+
+// for more information, see https://github.com/iter8-tools/iter8/issues/1257
+
+service ABN {
+  // Identify a version the caller should send a request to.
+  // Should be called for each request (transaction).
+  rpc Lookup(Application) returns(Session) {}
+
+  // Write a metric value to metrics database.
+  // The metric value is explicitly associated with a list of transactions that contributed to its computation.
+  // The user is expected to identify these transactions.
+  rpc WriteMetric(MetricValue) returns (google.protobuf.Empty) {}
+}
+
+message Application {
+  // name of (backend) application or service
+  // This value is used to identify the Kubernetes objects that make up the service
+  // Kubernetes objects that comprise the service should have the label app.kubernetes.io/name set to name
+  string name = 1;
+  // User or user session identifier
+  string user = 2;
+}
+
+message Session {
+  // Track or logical name of the application version
```

Comment:
I have found it hard to define in a clear an understandable way the term "track'. I finally settled on the language "logical version". Is there a better way to define and describe this term?

Reply:
I did not understand that we wanted to restrict track names to a fixed static set. I did not implement it this way. Instead, the user may use any logical names they feel make sense to them. I can look at restricting the set of valid names.

Reply:
We want to fix track names, so that in the frontend code, the user can create a fixed track-to-route map. The frontend should not be affected when new versions of the backend are released. We achieving this by fixing the track -- and mapping tracks to routes (in the frontend, statically), and also mapping tracks to versions (in the abn service, dynamically, so that metrics can be associated with versions and not just tracks).

Reply:
Track = an identifier which can be statically mapped to a route to a backend version. The route can be an HTTP route (URL with headers and auth values) or a gRPC route (gRPC call implemented service) -- entirely up to the app (in particular, the frontend) to decide what a route means.

Reply:
Within the A/B/n service, there is nothing special that is required to fix the track names. The only thing that is required is to keep the concepts of track and versions distinct (as described above), so that we document the use of fixed track names.